typings/styled-components.d.ts

```diff
@@ -72,9 +72,7 @@ export interface ThemedCssFunction<T> {
 }
 
 // Helper type operators
-type Diff<T extends string, U extends string> = ({ [P in T]: P } & { [P in U]: never } & { [x: string]: never })[T];
-type Omit<T, K extends keyof T> = Pick<T, Diff<keyof T, K>>;
-type WithOptionalTheme<P extends { theme?: T; }, T> = Omit<P, "theme"> & { theme?: T; };
+type WithOptionalTheme<P extends { theme?: T; }, T> = Pick<P, Exclude<keyof P, "theme">> & { theme?: T; };
```

Comment:
if it's not going to be used anywhere else, then we can have only constraint `P extends { theme: T; }` (not optional)